library/Zend/InputFilter/Input.php

```diff
@@ -326,6 +326,10 @@ public function isValid($context = null)
             return true;
         }
 
+        if ($empty && !$this->isRequired()) {
```

Comment:
for example : 
```php
$foo = null;
if (!$foo) { echo "not"; } else { echo "yes"; }
```
produce 'not'.

but 

```php
$foo = null;
if ($foo === false) { echo "not"; } else { echo "yes"; }
```
produce 'yes'.

Well, It indeed problematic, I tried to provide a PR about it in last days, but it got many errors, as many codes should cast it to real (bool).